testflinger_agent/client.py

```diff
@@ -267,10 +292,42 @@ def post_agent_data(self, data):
         :param data:
             dict of various agent data points to send to the api server
         """
-        agent = self.config.get("agent_id")
         agent_data_uri = urljoin(self.server, "/v1/agents/data/")
-        agent_data_url = urljoin(agent_data_uri, agent)
+        agent_data_url = urljoin(agent_data_uri, self.config.get("agent_id"))
         try:
             self.session.post(agent_data_url, json=data, timeout=30)
         except RequestException as exc:
             logger.error(exc)
+
+    def post_influx(self, job_id, phase, duration, result):
+        """Post the relevant data points to testflinger server
+
+        :param data:
+            dict of various agent data points to send to the api server
+        """
+        data = [
+            {
+                "measurement": "phase result",
+                "tags": {
+                    "agent": self.config.get("agent_id"),
+                    "job_id": job_id,
```

Comment:
I have serious concerns about putting the job_id in as a tag. IIUC, tags are indexed metadata for influxdb. So if we have too many unique values stored for a tag, that's not going to be sustainable. In fact, it looks like influxdb has a [default max](https://archive.docs.influxdata.com/influxdb/v1.2/administration/config/#:~:text=max%2Dvalues%2Dper%2Dtag%20%3D%20100000&text=The%20default%20setting%20is%20100000,returns%20a%20partial%20write%20error.) of 100,000 values - after which you need to either reconfigure it or else it will no longer store values. We'll hit that REALLY fast for job IDs. Let's leave job_id out of the data - I don't think it's really needed here, and could actually be harmful to try to capture it

Reply:
Makes sense. One thing we could do is include job_id in the measurement field:
i.e.
measurement = '%s phase result' % job_id
...
"measurement": measurement

Reply:
I dunno, that could be ugly too, just a thought.

Reply:
It actually doesn't look too bad on the influx side via Chronograf. I'm not sure if the ordering of the strings makes sense, but it's a way to get context-specific measurements in place.

I'll push it out if you want to try it, if you don't think it's a good idea or makes sense then we can just remove it entirely.

Reply:
From a grafana standpoint, I'm not sure how we would use it though. job_id is relatively short-lived as jobs expire and get auto-deleted after some period of time. In that short duration where that data is alive, if someone really wants to know how long each phase takes, then I think we could simply add start/stop timestamps in the result data directly. However I'd really want to hear more about how they plan on using that data before implementing it. For now, I don't think I've heard any request for this though.